src/Microsoft.ML.Data/Evaluators/RankingEvaluator.cs

```diff
@@ -271,7 +270,7 @@ public sealed class Aggregator : AggregatorBase
         {
             public sealed class Counters
             {
-                public const int MaxTruncationLevel = 10;
+                public const int MaxTruncationLevel = 100;
```

Comment:
Is there a technical need for any fixed limit? From the data science side, I don't know of one. 

Is there a ramification in our code for simply removing the limit?

Reply:
The original author of this code has assumed 100 to be a reasonable limit and that assumption is enforced in many places in the code. The MaxTruncationLevel of 10 set above made it a bit inconsistent and my fix just makes the max limit consistent. 

In theory it should be possible to remove the limit. But the way the code is structured makes it a bit more invasive than necessary.

Right now, in `RankingEvaluator.cs` the static class `RankingUtils` class has a static method `DiscountMap` which carries this assumption with a hard coded constant of 100 and also has a comment to reflect that.

`//RankingEvaluator.cs on line 954
`var discountMap = new Double[100]; //Hard to believe anyone would set truncation Level higher than 100`

The way the code is structured, this requires some well defined maximum. It should be possible to remove this but that would require a bigger code change. I would like input on whether or the original author's assumption that 100 is a reasonable limit is a good assumption. If it is really necessary to not have a limit at all, it is a bigger code change. 

Can we take this fix as is and open a new bug to remove the limit and we can triage that separately?